src/coreComponents/fileIO/timeHistory/PackCollection.cpp

```diff
@@ -8,143 +8,211 @@ PackCollection::PackCollection ( string const & name, Group * parent )
   , m_objectPath( )
   , m_fieldName( )
   , m_setNames( )
-  , m_minimumSetSize( )
+  , m_setChanged( false )
+  , m_onlyOnSetChange( 0 )
+  , m_disableCoordCollection( false )
+  , m_initialized( false )
 {
-  registerWrapper( PackCollection::viewKeysStruct::objectPath, &m_objectPath ).
+  registerWrapper( PackCollection::viewKeysStruct::objectPathString(), &m_objectPath ).
     setInputFlag( InputFlags::REQUIRED ).
     setDescription( "The name of the object from which to retrieve field values." );
 
-  registerWrapper( PackCollection::viewKeysStruct::fieldName, &m_fieldName ).
+  registerWrapper( PackCollection::viewKeysStruct::fieldNameString(), &m_fieldName ).
     setInputFlag( InputFlags::REQUIRED ).
     setDescription( "The name of the (packable) field associated with the specified object to retrieve data from" );
 
-  registerWrapper( PackCollection::viewKeysStruct::setNames, &m_setNames ).
+  registerWrapper( PackCollection::viewKeysStruct::setNamesString(), &m_setNames ).
     setInputFlag( InputFlags::OPTIONAL ).
     setDescription( "The set(s) for which to retrieve data." );
 
-  registerWrapper( PackCollection::viewKeysStruct::minSetSize, &m_minimumSetSize ).
+  registerWrapper( PackCollection::viewKeysStruct::onlyOnSetChangeString(), &m_onlyOnSetChange ).
     setInputFlag( InputFlags::OPTIONAL ).
-    setDefaultValue( -1 ).
-    setDescription( "The minimum size of the set(s) to be collected (use for sets that expand during the simulation)." );
+    setDefaultValue( 0 ).
+    setDescription( "Whether or not to only collect when the collected sets of indices change in any way." );
 }
 
-void PackCollection::initializePostSubGroups()
+void PackCollection::initializePostSubGroups( )
 {
-  localIndex numSets = m_setNames.size( );
-  m_collectionCount = numSets == 0 ? 1 : numSets;
-  DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
-  updateSetsIndices( domain );
-  HistoryCollection::initializePostSubGroups();
+  if( !m_initialized )
+  {
+    DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
+    localIndex numSets = m_setNames.size( );
+    m_collectionCount = numSets == 0 ? 1 : numSets;
+    // determine whether we're collecting from a mesh object manager
+    Group const * targetObject = this->getTargetObject( domain, m_objectPath );
+    ObjectManagerBase const * objectManagerTarget = dynamic_cast< ObjectManagerBase const * >( targetObject );
```

Comment:
I always do like to make sure pointers are const if you don't plan on incrementing them.
```suggestion
    Group const * const targetObject = this->getTargetObject( domain, m_objectPath );
    ObjectManagerBase const * const objectManagerTarget = dynamic_cast< ObjectManagerBase const * >( targetObject );
```

...also we should make a templated version of `getTargetObject` that attempts the cast for you.